File: scrapers/theqoo.py
import httpx

from scrapers.base import BaseScraper, Post


class TheqooScraper(BaseScraper):
    community = "theqoo"
    community_name = "더쿠"
    base_url = "https://theqoo.net"

    async def fetch_best_posts(self, client: httpx.AsyncClient) -> list[Post]:
        soup = await self._fetch_html(client, f"{self.base_url}/hot")
        posts = []

        for row in soup.select("table.theqoo_board_table tbody tr"):
            tds = row.select("td")
            if len(tds) < 3:
                continue
            # 공지 스킵
            first = tds[0].get_text(strip=True)
            if "공지" in first or "이벤트" in first:
                continue

            a_tag = row.select_one("td.title a")
            if not a_tag:
                continue
            title = a_tag.get_text(strip=True)
            href = a_tag.get("href", "")
            if not href or not title:
                continue

            # 조회수: 마지막 td
            views = 0
            views_text = tds[-1].get_text(strip=True).replace(",", "")
            try:
                views = int(views_text)
            except ValueError:
                pass

            url = href if href.startswith("http") else f"{self.base_url}{href}"
            posts.append(self._make_post(title, url, views=views))

        return posts

File: internal/scraper/theqoo.go
package scraper

import (
	"fmt"
	"net/http"
	"strconv"
	"strings"

	"github.com/PuerkitoBio/goquery"
)

// TheqooScraper scrapes 더쿠 hot posts.
type TheqooScraper struct {
	baseScraper
}

func NewTheqooScraper() *TheqooScraper {
	return &TheqooScraper{
		baseScraper: baseScraper{
			community:     "theqoo",
			communityName: "더쿠",
			baseURL:       "https://theqoo.net",
			encoding:      "utf-8",
		},
	}
}

func (s *TheqooScraper) Name() string { return s.communityName }

func (s *TheqooScraper) FetchBestPosts(client *http.Client) ([]Post, error) {
	doc, err := fetchDocument(client, fmt.Sprintf("%s/hot", s.baseURL), s.encoding, s.baseURL)
	if err != nil {
		return nil, err
	}

	var posts []Post
	doc.Find("table.theqoo_board_table tbody tr").Each(func(_ int, row *goquery.Selection) {
		tds := row.Find("td")
		if tds.Length() < 3 {
			return
		}

		// Skip notices
		first := strings.TrimSpace(tds.First().Text())
		if strings.Contains(first, "공지") || strings.Contains(first, "이벤트") {
			return
		}

		aTag := row.Find("td.title a").First()
		if aTag.Length() == 0 {
			return
		}
		title := strings.TrimSpace(aTag.Text())
		href, exists := aTag.Attr("href")
		if !exists || href == "" || title == "" {
			return
		}

		// Views: last td
		views := 0
		lastTd := tds.Last()
		viewsText := strings.TrimSpace(strings.ReplaceAll(lastTd.Text(), ",", ""))
		if v, err := strconv.Atoi(viewsText); err == nil {
			views = v
		}

		url := href
		if !strings.HasPrefix(href, "http") {
			url = s.baseURL + href
		}
		posts = append(posts, s.makePost(title, url, 0, views, 0))
	})

	return s.filterPosts(posts), nil
}
